executor/aggregate.go

```diff
@@ -376,8 +373,14 @@ func (w *HashAggPartialWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitG
 		}
 		w.memTracker.Consume(-w.chk.MemoryUsage())
 		waitGroup.Done()
+		if w.stats != nil {
+			w.stats.PartialWorkerTime[index] = time.Since(start)
```

Comment:
How about creating a `aggWorkerStats` for `HashAggPartialWorker ` and `HashAggFinalWorker`?  
Here It will panic when `len(w.stats.PartialWorkerTime)` is 0.(I know normally it will not be 0...)

Reply:
But if `len(w.stats.PartialWorkerTime)` is 0, it means `HashAggPartialConcurrency` is 0.It will have no PartialWorker run() in this situation:-)